epimodel/gleam/batch.py

```diff
@@ -74,18 +74,10 @@ def open(cls, path):
         return cls(hdf, path, _direct=False)
 
     @classmethod
-    def new(cls, *, path=None, dir=None, comment=None):
+    def new(cls, *, path=None):
         """
         Create new batch HDF5 file.
-        Either `path` should be a (non-existing) file, or a `dir` should
-        be given - name is then auto-generated (with optional comment suffix).
         """
-        if path is None:
```

Comment:
This is a nice example of what is needed to do to make luigi works as expected. Because it needs to have knowledge before it runs it's logic (specified in the `run` method), I had to extract all "output path logic" out of this part so it's known when DAG is resolved.